sdks/RepoToolset/sdk/Sdk.targets

```diff
@@ -4,6 +4,35 @@
   <PropertyGroup>
     <MSBuildAllProjects>$(MSBuildThisFileFullPath);$(MSBuildAllProjects)</MSBuildAllProjects>
   </PropertyGroup>
+
+  <!--
   
-  <Import Project="..\tools\Imports.targets" Condition="'$(__ExcludeSdkImports)' != 'true'" />
+    When building the entire stack from source can only build projects that target .NET Core.
+    To avoid adding conditions to each projects we remove the unwanted targets here.
+    If the project does not target .NET Core at all we suppress its restore and build completely.
+  
+  -->
+
+  <PropertyGroup>
+    <_SuppressBuild>false</_SuppressBuild>
+    <_SuppressBuild Condition="'$(__ExcludeSdkImports)' == 'true'">true</_SuppressBuild>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(DotNetBuildFromSource)' == 'true' and !$(_SuppressBuild)">
+    <_ReducedTargetFrameworks Condition="'$(TargetFrameworks)' != ''">$([System.Text.RegularExpressions.Regex]::Replace('$(TargetFrameworks)', 'net[0-9]+;?', '').Trim(';'))</_ReducedTargetFrameworks>
+    <_ReducedTargetFrameworks Condition="'$(TargetFrameworks)' == ''">$([System.Text.RegularExpressions.Regex]::Replace('$(TargetFramework)', 'net[0-9]+', ''))</_ReducedTargetFrameworks>
```

Comment:
We want to keep all the net* targets though. RedHat was okay with us doing that as we're just compiling against the equivalent of header files here.

Reply:
I see. The projects that reference non-FX non-OSS assemblies need to be filtered out (anything that builds against VS assemblies, for example), right?

Reply:
Updated the filter to check for `ExcludeFromSourceBuild`. Projects can now set this property to exclude themselves from source build.